packages/core/src/Site/index.ts

```diff
@@ -571,10 +642,12 @@ class Site {
       this.variableProcessor.addUserDefinedVariable(base, 'MarkBind', MARKBIND_LINK_HTML);
 
       const $ = cheerio.load(content, { decodeEntities: false });
-      $('variable,span').each((index, element) => {
+      $('variable,span').each((_index, element) => {
         const name = $(element).attr('name') || $(element).attr('id');
 
-        this.variableProcessor.renderAndAddUserDefinedVariable(base, name, $(element).html());
+        if (name) {
```

Comment:
Could this result in a different behavior since this is a possible change in logic?

Reply:
Previously a non-user-facing error would be thrown if the user provides a variable without a name; this will instead silently ignore the variable that has no name.
We could either fail silently, or explicitly throw a user-facing error. Which do you think is better?

Reply:
Yup! I think logging a warning would be helpful. This would likely without specifying which file it came from though, since that would involve passing down a context object to wherever the addVariables method is called which may add a lot more complexity. Does that sound good?

Reply:
Sounds good!